### DISCUSSION TITLE
Authenticate user before critical actions

### DISCUSSION THREAD

**Discussion by iftachhh** [supabase/supabase#8701] [Questions]
Hey there, in my application I have number of critical actions as record deletion for example, which I want to prevent users from accidentally do. I has been wondering if I can take advantage of the auth feature, which I already rely on for authentication, to implement a sudo-mode-like confirmation? Thanks

**Comment by GaryAustin1** [comment]
Two ideas if you mean a mode to toggle on and off and check with RLS.

1) Use a "sudo" table or a profile column you update to set/reset before and after your "sudo mode".  Then in RLS you check the column for your critical functions.

2) Use a custom header and toggle it off and on by re-initializing the client with the header having a value. https://supabase.com/docs/reference/javascript/initializing#with-additional-parameters . This only works for PostgREST operations and you can check the header in RLS with:  https://postgrest.org/en/stable/api.html#http-context (this is similar to how auth RLS helper functions work, although they use the PostgREST jwt setting).

Advantage of 2nd approach is no extra table operation to update the mode, or extra table select in RLS to check it.
 
 if you need them to confirm password you could send that as the header and in RLS check it with encryption against their current password, or in the table case set the mode with an rpc call that checks the password and then sets a flag. In the table case you might need a way to shut off the mode with cron or signin.
